clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala

```diff
@@ -38,13 +39,15 @@ object GarbageCollector {
   ): Set[(String, Array[Byte], Array[Byte])] = {
     val location =
       new ApiClient(conf.apiURL, conf.accessKey, conf.secretKey).getMetaRangeURL(repo, commitID)
-    SSTableReader
-      .forMetaRange(new Configuration(), location)
-      .newIterator()
-      .map(range =>
-        (new String(range.id), range.message.minKey.toByteArray, range.message.maxKey.toByteArray)
-      )
-      .toSet
+    if (location == "") Set[(String, Array[Byte], Array[Byte])]()
```

Comment:
Why do you need the type signature here?  I would expect it to derive through the from the required return type.  E.g. this does type-check:
```scala
def f(x: Boolean): Set[(Int, Int)] = if (x) Set() else Set((2,3), (3,2))
```